extensions/upgradeinput/config.go

```diff
@@ -0,0 +1,30 @@
+package upgradeinput
+
+type PSACT string
+
+const (
+	ConfigurationFileKey = "upgradeInput" // ConfigurationFileKey is used to parse the configuration of upgrade tests.
+	localClusterID       = "local"        // localClusterID is a string to used ignore this cluster in comparisons
+	latestKey            = "latest"       // latestKey is a string to determine automatically version pooling to the latest possible
+)
+
+// Config is a struct that stores multiple clusters and their testing options to load from the configuration file
+type Config struct {
+	Clusters []Cluster `json:"clusters" yaml:"clusters" default:"[]"`
+}
+
+// Cluster is a struct that's used to configure a single cluster to be used in an upgrade test
+type Cluster struct {
+	Name              string   `json:"name" yaml:"name" default:""`
+	VersionToUpgrade  string   `json:"versionToUpgrade" yaml:"versionToUpgrade" default:""`
+	PSACT             string   `json:"psact" yaml:"psact" default:""`
+	FeaturesToTest    Features `json:"enabledFeatures" yaml:"enabledFeatures" default:""`
+	isLatestVersion   bool
```

Comment:
Shouldnt all fields and consts be made public so that they are accessible outside of the `upgradeinput` package? It seems like these are directly referenced in the current upgrade tests in the rancher/rancher repo.

Reply:
Yes, that's correct, that was the initial PR without rancher/rancher PR. Related ones are updated as public.